cpp/tests/reshape/interleave_columns_tests.cpp

```diff
@@ -693,6 +690,317 @@ TYPED_TEST(ListsColumnsInterleaveTypedTest, SlicedColumnsInputNullableChild)
+
+  auto const col1_original = ListsCol{
+    {
+      ListsCol{ListsCol{{null, 11}, null_at(0)},
+               ListsCol{{22, null, null}, nulls_at({1, 2})}},  // don't care
+      ListsCol{ListsCol{{null, 11}, null_at(0)},
+               ListsCol{{22, null, null}, nulls_at({1, 2})}},  // don't care
+      ListsCol{ListsCol{{null, 11}, null_at(0)},
+               ListsCol{{22, null, null}, nulls_at({1, 2})}},  // don't care
+      //
+      ListsCol{ListsCol{{null, 2, 3}, null_at(0)}, ListsCol{{4, null, null}, nulls_at({1, 2})}},
+      ListsCol{{ListsCol{7, 8}, ListsCol{9, 10}, ListsCol{null, null, null} /*NULL*/}, null_at(2)},
+      ListsCol{ListsCol{11, 12, 13}, ListsCol{{14, null}, null_at(1)}, ListsCol{16, 17}},
+      //
+      ListsCol{ListsCol{{null, 11}, null_at(0)},
+               ListsCol{{22, null, null}, nulls_at({1, 2})}}  // don't care
+    },
+    nulls_at({0, 2, 3})};
+  auto const col2_original = ListsCol{
+    ListsCol{ListsCol{{null, 11}, null_at(0)},
+             ListsCol{{22, null, null}, nulls_at({1, 2})}},  // don't care
+    ListsCol{ListsCol{{null, 11}, null_at(0)},
+             ListsCol{{22, null, null}, nulls_at({1, 2})}},  // don't care
+                                                             //
+    ListsCol{{ListsCol{11, 12, 13}, ListsCol{null, null} /*NULL*/}, null_at(1)},
+    ListsCol{ListsCol{17, 18}, ListsCol{{19, 110, null}, null_at(2)}},
+    ListsCol{ListsCol{111, 112, 13}, ListsCol{114, 115}, ListsCol{116, 117}},
+    ListsCol{ListsCol{{null, 11}, null_at(0)},
+             //
+             ListsCol{{22, null, null}, nulls_at({1, 2})}},  // don't care
+    ListsCol{ListsCol{{null, 11}, null_at(0)},
+             ListsCol{{22, null, null}, nulls_at({1, 2})}},  // don't care
+    ListsCol{ListsCol{{null, 11}, null_at(0)},
+             ListsCol{{22, null, null}, nulls_at({1, 2})}}  // don't care
+  };
+
+  auto const col1 = cudf::slice(col1_original, {3, 6})[0];
+  auto const col2 = cudf::slice(col2_original, {2, 5})[0];
+  auto const expected =
+    ListsCol{
+      {ListsCol{ListsCol{{null, 2, 3}, null_at(0)}, ListsCol{{4, null, null}, nulls_at({1, 2})}},
+       ListsCol{{ListsCol{11, 12, 13}, ListsCol{null, null} /*NULL*/}, null_at(1)},
+       ListsCol{{ListsCol{7, 8}, ListsCol{9, 10}, ListsCol{null, null, null} /*NULL*/}, null_at(2)},
+       ListsCol{ListsCol{17, 18}, ListsCol{{19, 110, null}, null_at(2)}},
+       ListsCol{ListsCol{11, 12, 13}, ListsCol{{14, null}, null_at(1)}, ListsCol{16, 17}},
+       ListsCol{ListsCol{111, 112, 13}, ListsCol{114, 115}, ListsCol{116, 117}}},
+      null_at(0)}
+      .release();
+  auto const results = cudf::interleave_columns(TView{{col1, col2}});
+  CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(*expected, *results, verbosity);
+}
+
+TYPED_TEST(ListsColumnsInterleaveTypedTest, InputListsOfStructsNoNull)
+{
+  using ColWrapper = cudf::test::fixed_width_column_wrapper<TypeParam, int32_t>;
+
+  auto structs1 = [] {
```

Comment:
I like this way of structuring `STRUCT` construction. I'll adopt this for my tests in the future.